### PR TITLE
[FIX] mrp: start the next workorder in a function to allow for overrides

### PR DESCRIPTION
**Pull request by len-foss**
When a workorder is processed, it starts the next workorder.
We put that logic in a function to allow other modules to override it.

opw 1943674

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_workorder.py [hunk 1]
```diff
@@ -409,6 +408,12 @@ def record_production(self):
             self.button_finish()
         return True
 
+    @api.multi
+    def _start_nextworkorder(self):
+        for record in self:
```
Comment (fmdl) [inline addons/mrp/models/mrp_workorder.py]: one write for better performance : `self.mapped('next_work_order_id').filtered(lambda x: x.state == 'pending').write({'state':'ready'})`

File: addons/mrp/models/mrp_production.py [hunk 2]
```diff
@@ -583,6 +583,7 @@ def _workorders_create(self, bom, bom_data):
             })
             if workorders:
                 workorders[-1].next_work_order_id = workorder.id
+                workorders[-1]._init_nextworkorder_states()
```
Comment (Whenrow) [inline addons/mrp/models/mrp_production.py]: it's already done line 580

Comment (len-odoo) [reply addons/mrp/models/mrp_production.py]: No, l580 we give it an initial value (for all but the first, pending) but if there is a batch_size of 0, then we want to allow workorders to start in parallel. 
That's why we go over the state once again, but using the info from workorders[-1] to decide what its state should really be (although it's only really used in mrp_workorder...).

Comment (Whenrow) [reply addons/mrp/models/mrp_production.py]: ok you are right

**Comment by len-odoo** [comment]
robodoo r+

**Comment by len-odoo** [comment]
robodoo rebase-ff

**Pull request closed without merging**